app/src/main/java/com/infomaniak/mail/ui/main/move/MoveAdapter.kt

```diff
@@ -178,35 +175,20 @@ class MoveAdapter @Inject constructor(
         }
     }
 
-    @SuppressLint("NotifyDataSetChanged")
     fun setFolders(newFolders: List<Folder>, newCurrentFolderId: String? = null) = runCatchingRealm {
-
-        foldersDiffer.submitList(newFolders)
-
-        setFoldersJob?.cancel()
-        setFoldersJob = globalCoroutineScope.launch {
-            newCurrentFolderId?.let { currentFolderId = it }
-            val newHasCollapsableFolder = newFolders.any { it.canBeCollapsed }
-
-            val isFirstTime = hasCollapsableFolder == null
-            val collapsableFolderExistenceHasChanged = newHasCollapsableFolder != hasCollapsableFolder
-            hasCollapsableFolder = newHasCollapsableFolder
-
-            if (!isFirstTime && collapsableFolderExistenceHasChanged) {
-                Dispatchers.Main { notifyDataSetChanged() }
-            }
-        }
+        newCurrentFolderId?.let { currentFolderId = it }
+        submitList(newFolders)
     }
 
     fun updateSelectedState(newCurrentFolderId: String) {
         val previousCurrentFolderId = currentFolderId
         currentFolderId = newCurrentFolderId
-        previousCurrentFolderId?.let(::notifyCurrentItem)
-        notifyCurrentItem(newCurrentFolderId)
+        previousCurrentFolderId?.let(::notifyFolder)
+        notifyFolder(newCurrentFolderId)
```

Comment:
Since these two notify occur at the very same place and nowhere else, we could've looped through the list a single time and detected both at the same time to avoid looping over every folder twice. Is it worth it? I don't know

Reply:
This code was already there and is not used until the next menu drawer PR so it's fixed there